pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensible/channel/ServiceUnitState.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.loadbalance.extensible.channel;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Defines the possible states for service units.
+ *
+ * The following diagram defines the valid state changes
+ *
+ *                  ┌───────────┐
+ *       ┌──────────┤ released  │◄────────┐
+ *       │own       └───────────┘         │release
+ *       │                                │
+ *       │                                │
+ *       ▼                                │
+ *    ┌────────┐    assign          ┌─────┴────┐
+ *    │        ├───────────────────►│          │
+ *    │ owned  │                    │ assigned │
+ *    │        │◄───────────────────┤          │
+ *    └──┬─────┤      own           └──────────┘
+ *       │  ▲  │                         ▲
+ *       │  │  │                         │
+ *       │  │  └──────────────┐          │
+ *       │  │                 │          │
+ *       │  │        unload   │          │ assign
+ * split │  │                 │          │
+ *       │  │                 │          │
+ *       │  │ create(child)   │          │
+ *       │  │                 │          │
+ *       ▼  │                 │          │
+ *    ┌─────┴─────┐           └─────►┌───┴──────┐
+ *    │           │                  │          │
+ *    │ splitting ├────────────────► │not-owned │
+ *    │           │   discard(parent)│          │
+ *    └───────────┘                  └──────────┘
+ */
+public enum ServiceUnitState {
+
+    Owned, // the ownership assignment is complete (terminal state)
+
+    Assigned, // the ownership is assigned(but the assigned broker has not taken the ownership yet)
+
+    Released, // the ownership is released (e.g. the topic connections are closed)
+
+    Splitting; // the service unit(e.g. bundle) is in the process of splitting.
+    private static Map<ServiceUnitState, Set<ServiceUnitState>> validTransitions = new HashMap<>() {{
+        put(null, new HashSet<>() {{
```

Comment:
I'd prefer immutable sets over `HashSet` here. For example, `Set.of(...)`.

Reply:
Updated.